syntaxes/lean.json

```diff
@@ -55,23 +55,27 @@
     { "match": "\\b([0-9]+|0([xX][0-9a-fA-F]+))\\b", "name": "constant.numeric.lean" }
   ],
   "repository": {
-    "blockComment": {
-      "begin": "/-[^-]", "end": "-/", "name": "comment.block.lean",
-      "patterns": [{"include": "#blockComment"}]
-    },
-    "docComment": {
-      "begin": "/--", "end": "-/", "name": "comment.block.documentation.lean"
-    },
     "dashComment": {
       "begin": "(--)", "end": "$",
       "beginCaptures": {"0": {"name": "punctuation.definition.comment.lean"}},
       "name": "comment.line.double-dash.lean"
     },
+    "docComment": {
+      "begin": "/--", "end": "-/", "name": "comment.block.documentation.lean"
+    },
+    "modDocComment": {
+      "begin": "/-!", "end": "-/", "name": "comment.block.documentation.lean"
```

Comment:
I've left the "name" as "comment.block.documentation.lean" for now as I haven't yet figured out what the difference between "mod doc" and "doc" block comments yet.

Reply:
Module documentation comments come at the beginning of a file.  I don't think they are used anywhere at all.

Reply:
Ah, so "mod" stands for "module"! Makes sense.